spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/KerasLayer.scala

```diff
@@ -149,33 +153,52 @@ abstract class KerasLayer[A <: Activity: ClassTag, B <: Activity: ClassTag, T: C
     labor.accGradParameters(input, gradOutput)
   }
 
-  override def isCompatibleWithKeras(): Boolean = true
+  override def isBuilt(): Boolean = {
+    !this.modules.isEmpty && super.isBuilt()
+  }
 
-  override def isCompatibleWithTorch(): Boolean = false
+  override def isKerasStyle(): Boolean = true
 
-  override def getInputShape(): Shape = {
-    if (batchInputShape != null) {
-      batchInputShape
-    } else if (this.labor == null) {
-      null
-    } else {
-      this.labor.getInputShape()
+  override def computeOutputShape(inputShape: Shape): Shape = {
```

Comment:
In the previously design, we've implement lots of `computeOutputShape` in torch layers in case we want to support "InferShape" for Torch.
This is just a default implementation, the new added Keras layer can depend on this if it meets the requirement or override it with it's own logic.

Reply:
It's not clear to me if the default behavior is correct

Reply:
would simplify this logic once we introduce `KerasLayerWrapper`